src/org/zaproxy/zap/extension/custompages/CustomPageType.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2017 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.custompages;
+
+import org.apache.commons.lang.Validate;
+import org.parosproxy.paros.Constant;
+
+/**
+ * Defines the types supported by the Custom Page package.
+ */
+public enum CustomPageType {
+	ERROR(1, Constant.messages.getString("custompages.type.error")), 

```

Comment:
This one would be for 500 and the following for 404, the "other" would be a 200?

Reply:
Basically. I didn’t want to link them strongly to a particular response code. For “other” I didn’t have a planned mapping or much of an idea at all aside from giving othe devs options for using the module :)

Reply:
I'd prefer have them mapped to specific status codes (either by docs or name) as that's what the scanners check/need(?).

Reply:
So like `ERROR_500`, `NOTFOUND_404`, and `OTHER`. The point of including "other" was to give scripters etc a way to define something that doesn't fit a preconceived idea, that probably wouldn't be used by any of our packaged scanners (rules), then they can decide how to handle it under whatever given circumstance they might have. 

I'm ok with including a status code (number) in the naming if that helps/makes sense.